rdt-open-reader/src/testCovidIndonesia/java/io/ona/rdt/widget/GoogleCovidRDTBarcodeFactoryTest.java

```diff
@@ -0,0 +1,38 @@
+package io.ona.rdt.widget;
+
+import android.content.Intent;
+
+import com.google.android.gms.vision.barcode.Barcode;
+import com.vijay.jsonwizard.constants.JsonFormConstants;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.powermock.reflect.Whitebox;
+
+import io.ona.rdt.robolectric.widget.WidgetFactoryRobolectricTest;
+
+public class GoogleCovidRDTBarcodeFactoryTest extends WidgetFactoryRobolectricTest {
+
+    private GoogleCovidRDTBarcodeFactory googleCovidRDTBarcodeFactory;
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        googleCovidRDTBarcodeFactory = new GoogleCovidRDTBarcodeFactory();
+    }
+
+    @Test
+    public void testGetBarcodeValuesAsCSV() throws Exception {
+        Barcode barcode = new Barcode();
+        barcode.displayValue = "openrdt.ona.io.widget.GoogleCovidRDTBarcodeFactory\u001D31012299564524\u001D52605,M017G71\u001DMalariaPfPv\u001D5060511890000";
```

Comment:
Could we use `openrdt.ona.io/,31012021,52605,M017G71,MalariaPfPv,5060511890000` instead?

Reply:
chunks are too short for that reason the `extractValuesFromCSV` method thrown `StringIndexOutOfBoundsException` exception

Reply:
ok, could you use the value https://github.com/onaio/rdt-standard/issues/302#issue-643201478 instead?

Reply:
still can't use data breaking look like
```java
String[] dataSections = values.split(",");
String gtin = dataSections[0].substring(2, 16);
String lotNum = dataSections[0].substring(18);
String expDate = dataSections[1].substring(2, 8);
String uniqueId = dataSections[1].substring(10);
String tempSensor = dataSections[2];
```

Reply:
Could you figure out why? This is the expected barcode format.

Reply:
this method called.
https://github.com/onaio/rdt-standard/blob/00349b1e2ceea25aa074e25c620d3becc7f4d518/rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/widget/GoogleCovidRDTBarcodeFactory.java#L37-L51
inside the method of `splitCSV(String barcodeCSV)`
https://github.com/onaio/rdt-standard/blob/00349b1e2ceea25aa074e25c620d3becc7f4d518/rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/widget/GoogleCovidRDTBarcodeFactory.java#L25-L27

Reply:
Why is the exception happening? The issue I shared shows how the data is extracted. What makes the string invalid when the extraction code is considered?

Reply:
fixed, i was using the wrong splitter